include/cantera/base/Units.h

```diff
@@ -77,6 +81,50 @@ class Units
 };
 
 
+//! Unit aggregation utility
+/*!
+ *  Provides functions for updating and calculating effective units from a stack
+ *  of unit-exponent pairs. Matching units are aggregated, where a standard unit
+ *  simplifies access when joining exponents. The utility is used in the context
+ *  of effective reaction rate units.
+ */
+struct UnitStack
+{
+    UnitStack(const Units& standardUnits) {
+        stack.reserve(2); // covers memory requirements for most applications
+        stack.emplace_back(standardUnits, 0.);
+    }
+
+    //! Alternative constructor allows for direct assignment of vector
+    UnitStack(std::initializer_list<std::pair<Units, double>> units)
+        : stack(units) {}
+
+    //! Size of UnitStack
+    int size() const { return stack.size(); }
+
+    //! Get standard unit used by UnitStack
+    Units standardUnits() const;
+
+    //! Set standard units
+    void setStandardUnits(Units& standardUnits);
+
+    //! Effective exponent of standard unit
+    double standardExponent() const;
+
+    //! Join exponent of standard units
+    void join(double exponent);
```

Comment:
I'm having trouble parsing what "join" means here -- this could use a bit of elaboration in the documentation. It's effectively the same as "update" but only for the term that is defined as the "standard units", right?

Reply:
Yes, it's similar to `update`, but I didn't like the potential confusion of `updateStandard` (which could imply updating the *units*). Will update the docstring, for lack of better ideas ('aggregate' is an alternative). As it is now officially 'experimental', this can be changed at a later point.